client/landing/stepper/declarative-flow/copy-site.tsx

```diff
@@ -33,10 +72,11 @@ const copySite: Flow = {
 			recordFullStoryEvent( 'calypso_signup_start_copy_site', { flow: this.name } );
 		}, [] );
 
-		const urlQueryParams = useQuery();
-		const sourceSlug = urlQueryParams.get( 'sourceSlug' );
-		if ( ! sourceSlug ) {
-			window.location.assign( `/sites` );
+		const { isValidSite, hasFetchedSiteDetails, isFetchingError } = useIsValidSite();
+		if ( ! isValidSite ) {
+			if ( hasFetchedSiteDetails || isFetchingError ) {
+				window.location.assign( `/sites` );
+			}
 			return [];
```

Comment:
A nice enhancement would be adding a `Loading` state here.
- I created this issue as follow up https://github.com/Automattic/dotcom-forge/issues/1561